src/utils.py

```diff
@@ -1,109 +1,301 @@
-                    data_dict[subject][sample]['myelin'] = str(myelin_seg)
-        #add pixel_size
-        json_sidecar = next((data_path / subject / 'micr').glob('*.json'))
-        with open(json_sidecar, 'r') as f:
-            sidecar = json.load(f)
-        data_dict[subject]['sidecar'] = sidecar["PixelSize"][0]
-
-    return data_dict
-
-def adjust_pixel_values(img, pixel_size):
-    """Scales pixel values based on the provided pixel size."""
-    target_pixel_size = 0.1
-    scale_factor = pixel_size / target_pixel_size
-    img = (img / scale_factor).astype(np.uint8)
-    return img
-
-def load_bids_image(image_path, pixel_size_metadata=None):
-    """Loads a BIDS-formatted image and adjusts pixel values if needed."""
-    img = cv2.imread(image_path)
-    if pixel_size_metadata:
-        img = adjust_pixel_values(img, pixel_size_metadata)
-    return img
-
-def resize_and_pad(img, target_size=(416, 416)):
-    """Resizes an image while maintaining aspect ratio and pads to target size."""
-    old_size = img.shape[:2]
-    ratio = min(target_size[0] / old_size[0], target_size[1] / old_size[1])
-    new_size = tuple([int(x * ratio) for x in old_size])
-
-    img = cv2.resize(img, (new_size[1], new_size[0]))
-
-    delta_w = target_size[1] - new_size[1]
-    delta_h = target_size[0] - new_size[0]
-    top, bottom = delta_h//2, delta_h-(delta_h//2)
-    left, right = delta_w//2, delta_w-(delta_w//2)
-
-    img = cv2.copyMakeBorder(img, top, bottom, left, right, cv2.BORDER_CONSTANT, value=[0, 0, 0])
-    return img
-
-def normalize_and_window(img):
-    """Normalizes and applies intensity windowing to an image."""
-    img = (img - img.min()) / (img.max() - img.min())  # Normalize to [0, 1]
-    low_percentile, high_percentile = np.percentile(img, (2, 98))
-    img = np.clip(img, low_percentile, high_percentile)
-    img = (img - img.min()) / (img.max() - img.min())  # Renormalize
-    img = (img * 255).astype(np.uint8)  # Scale to [0, 255]
-    return img
-
-def find_regions(img):
-    """Finds connected regions directly from a binary segmentation mask."""
-    # Ensure the image is of integer type for labeling
-    img = img.astype(int)
-
-    # Label connected regions
-    labeled_img = measure.label(img)
-
-    # Extract region properties
-    regions = measure.regionprops(labeled_img)
-    if len(regions) == 0:
-        print("No regions found!")
-
-    return regions
-
-def crop_and_resize(img, region, target_size=(416, 416)):
-    """Crops an image around a region and resizes it."""
-    if region.area == 0:
-        print("Region area is zero!")
-        return np.zeros((target_size[0], target_size[1], 3), dtype=np.uint8)  # Return a black image
-
-    minr, minc, maxr, maxc = region.bbox
-    cropped_img = img[minr:maxr, minc:maxc]
-    return resize_and_pad(cropped_img, target_size)
+from tqdm import tqdm
+
+from utils import data_split
+
+SEM_DATASET_URL = "https://github.com/axondeepseg/data_axondeepseg_sem"

```

Comment:
It might be a good idea here to point to either a specific commit or specific version (if there are releases), in case there are changes in the future that are incompatible with this code.